code/sdk-bom/build.gradle

```diff
@@ -0,0 +1,211 @@
+    implementation 'com.adobe.marketing.mobile:media:latest.release'
+    implementation 'com.adobe.marketing.mobile:target:latest.release'
+    implementation 'com.adobe.marketing.mobile:campaign:latest.release'
+    implementation 'com.adobe.marketing.mobile:audience:latest.release'
+}
+
+configurations.implementation.canBeResolved = true
+
+task storeLatestExtensionInfo(type: WriteProperties) {
+    outputFile './bom.properties'
+    properties(retrieveLatestExtensionInfo())
+}
+
+Map<String, String> retrieveLatestExtensionInfo() {
+    def map = [:]
+    configurations.implementation.resolvedConfiguration.resolvedArtifacts.each {
+        def group = it.moduleVersion.id.group
+        def name = it.moduleVersion.id.name
+        def version = it.moduleVersion.id.version
+        if (group == "com.adobe.marketing.mobile") {
+            map["${group}-${name}"] = version
+        }
+    }
+    return map
+}
+
+task testIt() {
+    doLast {
+        println(retrieveLatestExtensionInfo())
+        println(retrieveExtensionVersionFromBomProperties())
+    }
+}
+
+Map<String, String> retrieveExtensionVersionFromBomProperties() {
+    Map<String, String> map = [:]
+    def props = new Properties()
+    file("bom.properties").withInputStream { props.load(it) }
+    props.each {
+        def key = it.key.toString()
+        def value = it.value
+        if (key.startsWith("com.adobe.marketing.mobile")) {
+            def extensionName = key.replace("com.adobe.marketing.mobile-", "")
+            map[extensionName] = value
+        }
+    }
+    return map
+}
+
+
+task computeSdkSize() {
+    doLast {
+        def formatStr = "%,10.2f"
+        def size = configurations.implementation.collect { it.length() / (1024 * 1024) }.sum()
+        def out = new StringBuffer()
+        if (size) {
+            out << "\n"
+            out << "-------------------------------------------------------\n"
+            out << 'Android SDK'.padRight(40)
+            out << "${String.format(formatStr, size)} Mb\n"
+            out << "-------------------------------------------------------\n"
+
+            configurations.implementation.sort { -it.length() }
+                    .each {
+                        if (it.name.contains("kotlin-stdlib") || it.name.contains("constraintlayout") || it.name.contains("annotations")) {
+                            return true
+                        }
+                        def items = it.name.replace(".aar", "").split("-")
+                        out << "${items[0]}".padRight(25)
+                        out << "${items[1]}".padRight(15)
+                        out << "${String.format(formatStr, (it.length() / 1024))} kb\n"
+                    }
+        } else {
+            out << 'No dependencies found';
+        }
+        out << "-------------------------------------------------------\n"
+        println(out)
+    }
+}
+
+task bumpBomMinorVersion(){
```

Comment:
Could we create a script that automatically adjusts the version number based on changes to the extension versions, rather than consistently incrementing the minor version? 

- Major version increase for addition, removal, or major version update of extensions.
- Minor version increase for minor version updates of extensions.
- Patch version increase for patch updates to extensions.

Reply:
Good idea! I will update the Gradle task soon.